src/lib.rs

```diff
@@ -644,6 +654,15 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         Ok(self.0.tls_server_end_point()?)
     }
 
+    /// Returns the negotiated ALPN protocols
+    pub fn negotiated_alpn(&self) -> Result<Option<String>> {
+        let protocol = self
+            .0
+            .negotiated_alpn()?
+            .map(|bytes| String::from_utf8_lossy(&bytes).into_owned());
```

Comment:
We shouldn't be transparently changing the values received from the server either. This should probably just return `Vec<u8>`.

Reply:
Fair enough changed it back.